src/value.rs

```diff
@@ -215,6 +217,40 @@ impl Uninitialized for Value {
     }
 }
 
+impl FromGlibPtrNone<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut gobject_ffi::GValue) -> Self {
+        assert!(!ptr.is_null());
+        Value(ptr::read(ptr))
```

Comment:
This causes two owners of the same memory if the GValue is some kind of pointer (object, string, boxed type, ...). Both your new Value and the C GValue will contain the same pointer, and both think they own it and will try to free it later, leaving the other one with a pointer to a now invalid memory address.